src/object_.d

```diff
@@ -253,6 +259,16 @@ class TypeInfo
         auto ti = cast(const TypeInfo)o;
         return ti && this.toString() == ti.toString();
     }
+    bool opEquals(const TypeInfo ti)
```

Comment:
TypeInfo's derivatives are all defined in druntime. If none of them define opEquals, then I think it's safe to make this final. In any case, it's an implementation detail that can be changed later.